### PR TITLE
fix(react-native): fix blank video/audio streams and muting state

### PR DESCRIPTION
**Pull request by santhoshvai**
Fixes the following issues: 

- muting state is now a global state.. (so when leave a call etc, it is still persisted).. also the persistence can be removed at any point by a component anywhere..
    how it should have been? muting state must be local to a lifecycle of call.. aka from lobbyview to active call, or incoming call to active call
- we initially publish streams without checking if call has been joined.. essentially creating a random race.. wherein we might randomly publish initially before the join happened and thereby streams wont be visible

### CODE REVIEW

File: packages/react-native-sdk/src/providers/MediaStreamManagement/index.tsx
Comment (vanGalilea) [inline packages/react-native-sdk/src/providers/MediaStreamManagement/index.tsx:1]: Very similar to the React [one](https://github.com/GetStream/stream-video-js/blob/4286d9855cc85c3f0f05d0d19b75e423c55591ce/packages/react-sdk/src/core/contexts/MediaDevicesContext.tsx#L234)

Comment (vanGalilea) [reply packages/react-native-sdk/src/providers/MediaStreamManagement/index.tsx:1]: Can't we know not benefit out of it and move it to the bindings?

File: packages/react-native-sdk/src/providers/MediaStreamManagement/useAudioPublisher.ts [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+   * 1. joining the call from the lobby, and the video is not muted
+   * 2. reconnecting to the call with the video already published
+   * 3. when the video device is changed (this is handled by the dependency to publishVideoStream function)
+   */
+  useEffect(() => {
+    if (callingState === CallingState.JOINED) {
+      if (
+        (!initialAudioMuted && !initialPublishRun.current) ||
+        isPublishingAudio
+      ) {
```
Comment (vanGalilea) [inline packages/react-native-sdk/src/providers/MediaStreamManagement/useAudioPublisher.ts:73]: can we merge the if statements or take the early exit approach you already took above?

Comment (santhoshvai) [reply packages/react-native-sdk/src/providers/MediaStreamManagement/useAudioPublisher.ts]: done!

File: packages/react-native-sdk/src/providers/MediaStreamManagement/index.tsx
Comment (santhoshvai) [reply packages/react-native-sdk/src/providers/MediaStreamManagement/index.tsx:1]: No we cannot, it has more browser specific stuff.. 

like extra states for video inputs, listeners for device disconnection

Comment (santhoshvai) [reply packages/react-native-sdk/src/providers/MediaStreamManagement/index.tsx:1]: and we add native permission checkers

Comment (santhoshvai) [reply packages/react-native-sdk/src/providers/MediaStreamManagement/index.tsx:1]: also in web they use default device id for video.. which is not relevant for rn-webrtc

Comment (vanGalilea) [reply packages/react-native-sdk/src/providers/MediaStreamManagement/index.tsx:1]: 👍🏻

**Review by vanGalilea — APPROVED** [review]

File: packages/react-native-sdk/src/providers/MediaStreamManagement/useVideoPublisher.ts
[the remark this answers is not in the document]
Comment (santhoshvai) [reply packages/react-native-sdk/src/providers/MediaStreamManagement/useVideoPublisher.ts:1]: No in my view, reasons:

-- keeping close to react
-- we will need to do some special stuff for video in reconnection flow, like republishing when coming back to foreground
-- cleaner with separation, although its similar you can reuse any.. so separation makes it easier to maintain

**Comment by vanGalilea** [comment]
Tested, no visible regression throughout main flow and E2E tests ✅

**Pull request merged**